libraries/libfc/src/log/gelf_appender.cpp

```diff
@@ -117,9 +130,17 @@ namespace fc
 
       if (my->gelf_endpoint)
       {
-        my->gelf_socket.initialize(io_service);
+        my->gelf_socket.initialize(my->io_context);
         my->gelf_socket.open();
         std::cerr << "opened GELF socket to endpoint " << my->cfg.endpoint << "\n";
+
+        my->thread = std::thread([this] {
+           try {
+              fc::set_os_thread_name("gelf_appender");
+              my->io_context.run();
+           } catch (...) {
```

Comment:
Should `elog` here at the very least.

Reply:
Obviously, as @huangminghuang pointed out we don't want to use `elog` as it would be recursive. How about adding a `std::cerr` call.